src/main/java/com/example/bookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.example.bookstore.service.impl;
+
+import com.example.bookstore.dto.user.UserRequestDto;
+import com.example.bookstore.dto.user.UserResponseDto;
+import com.example.bookstore.exception.RegistrationException;
+import com.example.bookstore.mapper.UserMapper;
+import com.example.bookstore.repository.user.UserRepository;
+import com.example.bookstore.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserMapper userMapper;
+    private final UserRepository userRepository;
+
+    public UserServiceImpl(UserMapper userMapper, UserRepository userRepository) {
+        this.userMapper = userMapper;
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserResponseDto register(UserRequestDto requestDto) throws RegistrationException {
+        if (userRepository.existsByEmail(requestDto.email())) {
+            throw new RegistrationException("Can't register user. Email is taken");
```

Comment:
add email to message
It's better: User with this email: already exists

Reply:
I don't feel like adding users credentials to the error messages.

Reply:
I didn't mention password that should be secured